predicators/main.py

```diff
@@ -299,6 +302,17 @@ def _run_testing(env: BaseEnv, approach: BaseApproach) -> Metrics:
             solved = task.goal_holds(traj.states[-1])
             exec_time = execution_metrics["policy_call_time"]
             metrics[f"PER_TASK_task{test_task_idx}_exec_time"] = exec_time
+            # Save the successful trajectory, e.g., for playback on a robot.
+            traj_file = f"{save_prefix}__task{test_task_idx+1}.traj"
+            traj_file_path = Path(CFG.eval_trajectories_dir) / traj_file
+            # Include the original task too so we know the goal.
+            traj_data = {
+                "task": task,
+                "trajectory": traj,
+                "pybullet_robot": CFG.pybullet_robot
+            }
+            with open(traj_file_path, "wb") as f:
+                pkl.dump(traj_data, f)
```

Comment:
So the plan is to have a script, that given any task, trajectory, and pybullet robot will generate the LISDF plan right?

Reply:
In the very long run we may want to output a LISDF input problem as well so we can load it dynamically on multiple simulators and its more self-contained.

Reply:
right, and yeah agreed!